### PR TITLE
libcava + cava application

### PR DESCRIPTION
**Pull request by LukashonakV**
Hi @karlstav , finally I got it :)
This PR makes:
1. Split necessary API into separate functions
2. Provide an opportunity to build simultaneously in one shot both: cava application + libcava
3. I've checked: both cava works well + Waybar now is able to use libcava without necessity to have cava as a subproject
4. I aligned the freshest cava from the master with the current branch
----
1. To build: make clean & make
2. To install: make install

### CODE REVIEW

File: src/cavacore.c [hunk 1]
```diff
@@ -523,7 +524,7 @@ void cava_execute(double *cava_in, int new_samples, double *cava_out, struct cav
             p->sens_init = 0;
         } else {
             if (!silence) {
-                p->sens = p->sens * 1.002;
+                p->sens = p->sens * 1.01;
```
Comment (LukashonakV) [inline src/cavacore.c:527]: Possible this change I didn't find during refactoring. Need to roll back once PR is approved

File: src/config.c [hunk 2]
```diff
@@ -783,9 +787,15 @@ bool load_config(char configPath[PATH_MAX], struct config_params *p, bool colors
     }
     GetPrivateProfileString("input", "source", "auto", p->audio_source, 64, configPath);
 
-    if (strcmp(outputMethod, "sdl") == 0 || strcmp(outputMethod, "sdl_glsl") == 0) {
```
Comment (LukashonakV) [inline src/config.c:786]: Again, looks like these changes I also didn't notice during refactoring. Possible need to rollback once PR is approved to merge

File: src/config.c [hunk 3]
```diff
@@ -783,9 +787,15 @@ bool load_config(char configPath[PATH_MAX], struct config_params *p, bool colors
...
+    if (strcmp(outputMethod, "sdl") == 0) {
         p->bar_width = GetPrivateProfileInt("general", "bar_width", 20, configPath);
         p->bar_spacing = GetPrivateProfileInt("general", "bar_spacing", 5, configPath);
+    }
```
Comment (LukashonakV) [inline src/config.c:793]: Again, looks like these changes I also didn't notice during refactoring. Possible need to rollback once PR is approved to merge

File: src/config.c [hunk 4]
```diff
@@ -783,9 +787,15 @@ bool load_config(char configPath[PATH_MAX], struct config_params *p, bool colors
...
+    if (strcmp(outputMethod, "sdl_glsl") == 0) {
+        p->bar_width = GetPrivateProfileInt("general", "bar_width", 1, configPath);
+        p->bar_spacing = GetPrivateProfileInt("general", "bar_spacing", 0, configPath);
+    }
+    if (strcmp(outputMethod, "sdl_glsl") == 0 || strcmp(outputMethod, "sdl") == 0) {
```
Comment (LukashonakV) [inline src/config.c:798]: Again, looks like these changes I also didn't notice during refactoring. Possible need to rollback once PR is approved to merge

File: src/input/pipewire.c [hunk 5]
```diff
@@ -33,7 +33,7 @@ static void on_process(void *userdata) {
     if ((samples = buf->datas[0].data) == NULL)
         return;
 
-    n_samples = buf->datas[0].chunk->size / (data->cava_audio->format / 8);
+    n_samples = buf->datas[0].chunk->size / sizeof(float);
```
Comment (LukashonakV) [inline src/input/pipewire.c:36]: Again, looks like these changes I also didn't notice during refactoring. Possible need to rollback once PR is approved to merge

File: src/input/pipewire.c [hunk 6]
```diff
@@ -18,8 +18,8 @@ static void on_process(void *userdata) {
     struct pw_data *data = userdata;
     struct pw_buffer *b;
     struct spa_buffer *buf;
+    float *samples;
     uint32_t n_samples;
-    int16_t *samples;
```
Comment (LukashonakV) [inline src/input/pipewire.c:23]: Again, looks like these changes I also didn't notice during refactoring. Possible need to rollback once PR is approved to merge

File: src/input/pipewire.c [hunk 7]
```diff
@@ -33,7 +33,7 @@ static void on_process(void *userdata) {
     if ((samples = buf->datas[0].data) == NULL)
         return;
 
-    n_samples = buf->datas[0].chunk->size / (data->cava_audio->format / 8);
+    n_samples = buf->datas[0].chunk->size / sizeof(float);
```
Comment (LukashonakV) [inline src/input/pipewire.c:37]: Again, looks like these changes I also didn't notice during refactoring. Possible need to rollback once PR is approved to merge

File: include/output/sdl_cava.h [hunk 8]
```diff
@@ -1,7 +1,7 @@
 #include "../config.h"
 
 void init_sdl_window(int width, int height, int x, int y);
-void init_sdl_surface(int *width, int *height, char *const fg_color_string,
+void init_sdl_surface(int *w, int *h, char *const fg_color_string,
                       char *const bg_color_string, int gradient, int gradient_count,
```
Comment (LukashonakV) [inline include/output/sdl_cava.h:5]: Again, looks like these changes I also didn't notice during refactoring. Possible need to rollback once PR is approved to merge

**Comment by karlstav** [comment]
This is a big change, I will have to do some thinking before I can consider merging it. 

First things you will need to fix the automake stuff so all the checks can run.

**Comment by LukashonakV** [comment]
Hi @karlstav ... ufff finally I stabilized code and:
1. Made make happy. Autotool is still able to build cava without any problems
2. I restructured project and separated necessary functionalities into APIS
3. I provided separate meson script which can be used when some project wants to integrate with cava through libcava
----
So... Users are able to build cava in old manner using autotools.

In order to build and install cava as a lib:
```bash
1. meson build
2. meson compile -C build
3. ninja -C build install
```

**Comment by karlstav** [comment]
I fear that the changes in this PR are bit too great that I can merge them at this time. I would recommend maintaining your own fork for now. Maybe see if there is some way of achieving your objective with fewer and more incremental changes.

**Comment by LukashonakV** [comment]
Ok.
By the way  libcava is living in the fork [libcava](https://github.com/LukashonakV/cava)

**Pull request closed without merging**